src/core/server/http/router/router.ts

```diff
@@ -52,40 +58,36 @@ export interface IRouter {
    * @param route {@link RouteConfig} - a route configuration.
    * @param handler {@link RequestHandler} - a function to call to respond to an incoming request
    */
-  get: <P extends ObjectType, Q extends ObjectType, B extends ObjectType>(
-    route: RouteConfig<P, Q, B>,
-    handler: RequestHandler<P, Q, B>
-  ) => void;
+  get: RouteRegistrar;
 
   /**
    * Register a route handler for `POST` request.
    * @param route {@link RouteConfig} - a route configuration.
    * @param handler {@link RequestHandler} - a function to call to respond to an incoming request
    */
-  post: <P extends ObjectType, Q extends ObjectType, B extends ObjectType>(
-    route: RouteConfig<P, Q, B>,
-    handler: RequestHandler<P, Q, B>
-  ) => void;
+  post: RouteRegistrar;
 
   /**
    * Register a route handler for `PUT` request.
    * @param route {@link RouteConfig} - a route configuration.
    * @param handler {@link RequestHandler} - a function to call to respond to an incoming request
    */
-  put: <P extends ObjectType, Q extends ObjectType, B extends ObjectType>(
-    route: RouteConfig<P, Q, B>,
-    handler: RequestHandler<P, Q, B>
-  ) => void;
+  put: RouteRegistrar;
 
   /**
    * Register a route handler for `DELETE` request.
    * @param route {@link RouteConfig} - a route configuration.
    * @param handler {@link RequestHandler} - a function to call to respond to an incoming request
    */
-  delete: <P extends ObjectType, Q extends ObjectType, B extends ObjectType>(
-    route: RouteConfig<P, Q, B>,
+  delete: RouteRegistrar;
+
+  /**
+   * Wrap a router handler to catch and converts legacy boom errors to proper custom errors.
+   * @param handler {@link RequestHandler} - a route handler to wrap
+   */
+  wrapErrors: <P extends ObjectType, Q extends ObjectType, B extends ObjectType>(
     handler: RequestHandler<P, Q, B>
-  ) => void;
+  ) => RequestHandler<P, Q, B>;
```

Comment:
Two questions here:

1. Should we flag this as deprecated right now, or should we wait?
2. Name is open to proposals. I don't think `wrapErrors` is explicit enough. `wrapLegacyErrors`? `convertLegacyErrors`? As we may handle more than only Boom at some point, I don't think we should go to something like `wrapBoomErrors`, but WDYT?

Reply:
How about `handleLegacyErrors`? I don't think `wrap` is right since it doesn't wrap an interface over the underlying Boom error. I also prefer `handle` as it implies that it does more than just convert the value from one to another since it actually catches and returns the error.